test/replay/integration/replayMap.test.js

```diff
@@ -0,0 +1,180 @@
+/**
+ * Integration tests for ReplayMap with API
+ */
+
+/* globals describe */
+/* globals it */
+/* globals beforeEach */
+/* globals afterEach */
+
+import { expect } from 'chai';
+import sinon from 'sinon';
+
+import ReplayMap from '../../../src/browser/replay/replayMap.js';
+import Recorder from '../../../src/browser/replay/recorder.js';
+import Tracing from '../../../src/tracing/tracing.js';
+import { SpanExporter } from '../../../src/tracing/exporter.js';
+import { spanExportQueue } from '../../../src/tracing/exporter.js';
+import Api from '../../../src/api.js';
+import { mockRecordFn } from '../util';
+
+const mockWindow = {
+  sessionStorage: {},
+  document: {
+    location: {
+      href: 'https://example.com/test',
+    },
+  },
+  navigator: {
+    userAgent: 'Mozilla/5.0 Test',
+  },
+};
```

Comment:
The tests run in the context of a headless chrome instance, and usually tests that interact with window state are better using the real browser window unless there's a specific reason to mock.

Example:
https://github.com/rollbar/rollbar.js/blob/master/test/tracing/tracing.test.js#L35

Reply:
Got it, will look into it, thanks.